docs/javascript_api.md

```diff
@@ -0,0 +1,37 @@
+# JavaScript API
+
+You can run Cucumber programmatically via its JavaScript API. This isn't something most users would need to do, but if you have a niche use case or need to work Cucumber into a bigger framework, it might be what you need. The API allows you to load configuration, sources and support code and execute test runs with async functions and within the current process.
+
+The API is available on its own entry point `@cucumber/cucumber/api`.
+
+There are some examples below, and [reference documentation](./api/index.md) for everything available.
+
+## Simple example
+
+```javascript
+import { loadConfiguration, runCucumber } from '@cucumber/cucumber/api'
+
+export async function runTests() {
+  const { runnable } = await loadConfiguration()
+  const { success } = await runCucumber(runnable)
+  return success
+}
+```
+
+## Complex example
+
+```javascript
+import { loadConfiguration, loadSupport, runCucumber } from '@cucumber/cucumber/api'
+
+export async function runTests(directory, configFile, failFast) {
+  // things we need to specify about the environment
+  const environment = { cwd: directory }
+  // load configuration from a particular file, and override a specific option
+  const { runnable } = await loadConfiguration({ file: configFile, provided: { failFast } }, environment)
```

Comment:
What does `runnable` mean?
(I guess I know what it means - I put myself as a user here ;))

Reply:
Right, so on the returned configuration object we have:

- `runnable` - the structured configuration object that can be passed to `runCucumber` (and can have the `support` field swapped out for an existing support code library)
- `original` the "flat" configuration that we've pulled together from a file (profiles) and whatever overrides were passed in - useful for debugging, and also has a couple of properties we don't include on `runnable` since they aren't used except in the CLI (`forceExit` and `publishQuiet`)

Maybe the fact I'm destructuring in the examples makes it confusing? It could change to like

```javascript
const configuration = await loadConfiguration()

const { success } = await runCucumber(configuration.runnable)
```

But also yeah, open to better names without being overly verbose. It's not too late (yet).

Reply:
Dunno if you saw my last edit

Yeah, I noticed the output of the loadConfiguration API.

For the documentation, indeed your snippet may be more clear.
But still, "runnable" for something that is actually not runnable (it does not have a "run" method)

What about one of the following, or something similar, to replace `originalConfiguration`
- `options`
- `userConfiguration`
- `requestedConfiguration`

and for `runnable`:
- `configuration`
- `cucumberConfiguration`
- `sanitizedConfiguration`

?

Reply:
I think `userConfiguration` and `runConfiguration` is a winner, I'll make that edit - thanks!

Reply:
(this is done now)

Reply:
Whoops! Fixed now :)

Reply:
Could `runConfiguration` be called a `testSuite`?